### DISCUSSION TITLE
Contributors wanted!

### DISCUSSION THREAD

**Discussion by StudentWeis** [ScoopInstaller/Extras#9291] [General]
There are two dozen PRs here that have not been processed. I do not know why

I love Scoop and hope it keeps getting better.

**Comment by amorphobia** [comment]
It's nearly 100 now.

**Comment by ivaquero** [comment]
Over 100 already.

**Comment by StarsbySea** [comment]
over 200 already 😂

**Comment by kriswilk** [comment]
Found this thread when I wondered why a simple PR (#10526) hadn't been reviewed/merged since mid-Feb. There seems to have been an absolute flood of new manifest PRs recently. Has there been some organized initiative to add all these new applications?

I don't have a problem waiting in line for review. But I've also noticed many newer PRs of a similar nature already merged. Perhaps just a case of the newest activity getting the attention....?

**Comment by HUMORCE** [comment]
> Perhaps just a case of the newest activity getting the attention....?

No, but fixes and improvements of manifests will be prioritised.

**Reply by kriswilk** [reply]
> > Perhaps just a case of the newest activity getting the attention....?
> 
> No, but fixes and improvements of manifests will be prioritised.

I agree, that is perfectly logical! I only wondered because of PRs like #10925 that was merged last week. It is based on request #10856 from only 3 weeks ago.

I also still wonder about the sudden massive influx of new manifest PRs (many from the same author). Have these all been individually/specifically requested, or has there been some kind of "farming/harvesting" of apps to add to scoop? It just seems like a lot all at once. And some seem rather obscure.

Just to be clear, I know yours is a virtually thankless job. I appreciate your effort!! Thank you! 👍👍

**Reply by HUMORCE** [reply]
Some PRs have other contributors testing, providing feedback, or reviewing within them, this means that there are people paying attention to these manifests.

**Comment by zsuatem** [comment]
Any chance of faster pull request merging? There is still a lot of stuff still from 2022 😭
I started to move what I want to have up to date to my bucket because, unfortunately, currently there is a very long wait for all requests.

I hope this will change because Scoop is for me the best of the three Scoop, Winget, Chocolatey and creating your own bucket just to have a new program or a newer version is lame 🫤

**Reply by HUMORCE** [reply]
> Any chance of faster pull request merging?

If a PR is working properly and meets expectations(what it's going to fix/resolve), leave your feedback in the PR.

**Comment by wickles** [comment]
There has not been a single new manifest merged in over 2 months. How can we improve the situation? Do you need more contributors to help review? How can we help or get involved? @HUMORCE it looks like you were the only maintainer left contributing for most of this year!

**Reply by wickles** [reply]
@rashil2000 can you help get the community sorted here? It looks like you might be the only maintainer left contributing to `main` (!)

**Reply by rashil2000** [reply]
Hi @wickles! 

The ScoopInstaller org as a whole gets a huge amount of activity each day, as such it's infeasible for a couple of people to maintain all repos. I primarily handle the Scoop core and Main bucket repos. 
We do need more bucket maintainers - especially for the Extras bucket since it's the largest one. I can add people who can review/test PRs for manifest fixes, new applications and for triaging issues.

**Reply by octaviordz** [reply]
Hi @rashil2000 not familiar with the ScoopInstaller process (just read a bit some time ago) but I could help with the "Extra bucket" in specific with the software that I use (installed by scoop) vscode and gitextensions for example.

**Reply by wickles** [reply]
@rashil2000 I could help with review here. I maintain some external scoop buckets already. 

Process seems straightforward. Is there additional testing and auditing that needs to be done for each manifest beyond the automated verification? Is there some admission criteria beyond having an issue with some indication of popularity by stars or some such?

**Reply by HUMORCE** [reply]
There didn't seem to be any active maintainers after I started traveling. I carry apple silicon mbp only, <del>so I will not reviewing PRs that need to be tested for now.</del> I merged some PRs that include feedback via community or LGTM(literally).

@rasa We need more maintainers to keep the rolling update buckets are working.

**Comment by pureby** [comment]
@rashil2000 I can only imagine how tough and laborious it is to maintain a huge and growing project like this with just a couple of people; big thanks for your hard work!

I'm not a software developer, but maybe I can help with a couple ideas:

1. **Clear, transparent and _complete_ documentation on manifests** + step-by-step instruction "for dummies".
I know many people (including myself) would like to contribute to the project, but simply have no idea where to start because the process of creating, reviewing and editing manifests is poorly documented and not at all transparent to an "average" user _(read: power user)._ I even know a couple **developers** who had trouble submitting **their own software** to Scoop, because they struggled to understand manifests, and referred to the process as "going through hell". 

2. **Calling out community for help.**
When doing `scoop update` or `scoop status`, Scoop could print a short text message asking community for help, e.g.:
`Scoop project needs more reviewers! Visit https://scoop.sh/contribute to see how you can help.`

There is no shame in **acknowledging the problem** and **asking for help,** and I'm sure many people, myself included, are willing to invest some of their free time in the project, if provided with **proper tools to do the job** (starting with a clear, complete and non-self-contradictory documentation). 

I think **proper documentation** will eliminate the problem at it's **root,** as I know for a fact that many, many people out there absolutely love and admire Scoop and have time to invest in the project but lack the skills of a professional software developer. Those skills should not be required for a simple manifest review.

**Comment by rasa** [comment]
Perhaps we need a plan on how to divy up the work? I currently maintain our Nirsoft bucket. @pureby @StudentWeis @kriswilk @wickles @zsuatem @octaviordz Perhaps volunteers can sign up to manage a portion of the Extras bucket? Say, if we get two volunteers/teams, they each take half the repo (split alphabetically). If three, they each take 1/3, etc. 

We could even have a GitHub Action auto-assign to each member/team.

**Reply by rashil2000** [reply]
Hi @aliesbelik - I've seen many contributions from you in the Main bucket, would you be interested in this?

**Reply by aliesbelik** [reply]
Hi @rashil2000, sounds interesting to me.
I can't guarantee the same level of engagement in future but if my occasional contributions and reviews (mostly pretty trivial) can be useful at the moment, I'd be happy to participate.

**Reply by rashil2000** [reply]
Thank you @aliesbelik! I've sent you an invite for this repository.

**Comment by rashil2000** [comment]
Hi all! 

I'd firstly like to apologize for being absent in the past few months - I started my first job and managing time has been a herculean effort. 
 
> 1. **Clear, transparent and _complete_ documentation on manifests** + step-by-step instruction "for dummies".
>    I know many people (including myself) would like to contribute to the project, but simply have no idea where to start because the process of creating, reviewing and editing manifests is poorly documented and not at all transparent to an "average" user _(read: power user)._ I even know a couple **developers** who had trouble submitting **their own software** to Scoop, because they struggled to understand manifests, and referred to the process as "going through hell".
> 2. **Calling out community for help.**
>    When doing `scoop update` or `scoop status`, Scoop could print a short text message asking community for help, e.g.:
>    `Scoop project needs more reviewers! Visit https://scoop.sh/contribute to see how you can help.`

I love these ideas - especially the second one - backlinking either to the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md) or the [Scoop Wiki](https://github.com/ScoopInstaller/Scoop/wiki) in the command's output will definitely increase visibility. 

I like @rasa's idea too - let us know if anyone's willing to help out and we will add you as contributor. 

I have pinned this discussion on this repo's Discussion page.

**Reply by StudentWeis** [reply]
Maybe I can provide some help

**Reply by rashil2000** [reply]
@StudentWeis I have added you as maintainer.

**Reply by StudentWeis** [reply]
Okay, I will try my best to fulfill my duties.

**Reply by cscnk52** [reply]
@rashil2000 I would like to provide some help, I have previous contribute to [Main](https://github.com/ScoopInstaller/Main/issues?q=is%3Apr%20author%3Acscnk52), [Extras](https://github.com/ScoopInstaller/Extras/issues?q=is%3Apr%20author%3Acscnk52), [Versions](https://github.com/ScoopInstaller/Versions/issues?q=is%3Apr%20author%3Acscnk52) and [dorado](https://github.com/chawyehsu/dorado/pulls?q=is%3Apr+author%3Acscnk52) bucket, and I am also maintaining my own [cetacea](https://github.com/cscnk52/cetacea) bucket, I want to be a Maintainer to help scoop better.

**Comment by jasongodev** [comment]
I've been volunteering for quite some time. What I experience as a bottleneck is that reviewers like me can only do as much. I can only check, recommend, and research what's the best for each of the apps. Most of the time the manifests are ready to be merged on the spot. But as a reviewer I can only triage them by putting my comments.

I think what we need are more maintainers or some other role like specific app maintainers that can merge manifests.

Another issue is that Scoop is centralized approval while the likes of Chocolatey has assigned maintainers for each apps. While Scoop's approach makes the buckets more consistent as per Scoop's guidelines, this centralized approach causes backlogs when maintainers are not available. In Chocolatey, the maintainers are closely related with their respective apps, some are members of their own dev team. This makes the apps individually updated and does not affect the whole ecosystem.

With that, maybe we can open the apps for orphaning. So avid users can volunteer to be a dedicated maintainer of that particular manifest. Most of the times these avid users also knows best when it comes to the quirks of the installation process and the changes in the download pages and hashes.

**Reply by rashil2000** [reply]
> I've been volunteering for quite some time. What I experience as a bottleneck is that reviewers like me can only do as much. I can only check, recommend, and research what's the best for each of the apps. Most of the time the manifests are ready to be merged on the spot. But as a reviewer I can only triage them by putting my comments.

The maintainer role (as mentioned above) gives you merge access, btw.

> Another issue is that Scoop is centralized approval while the likes of Chocolatey has assigned maintainers for each apps. While Scoop's approach makes the buckets more consistent as per Scoop's guidelines, this centralized approach causes backlogs when maintainers are not available. In Chocolatey, the maintainers are closely related with their respective apps, some are members of their own dev team. This makes the apps individually updated and does not affect the whole ecosystem.
> 
> With that, maybe we can open the apps for orphaning. So avid users can volunteer to be a dedicated maintainer of that particular manifest. Most of the times these avid users also knows best when it comes to the quirks of the installation process and the changes in the download pages and hashes.

I feel like there are too many applications in the buckets for this to be successful, even if we start now.

**Reply by Elkiwa** [reply]
This is something I've been wondering for a while. Have scoop considered utilizing [Github's CODEOWNERS](https://www.fullstory.com/blog/taming-github-codeowners-with-bots/) to alleviate part of the ownership problem? The system doesn't require every file to have a dedicated maintainer.

**Comment by rashil2000** [comment]
> This is something I've been wondering for a while. Have scoop considered utilizing [Github's CODEOWNERS](https://www.fullstory.com/blog/taming-github-codeowners-with-bots/) to alleviate part of the ownership problem? The system doesn't require every file to have a dedicated maintainer.

I don't think this'll help us much in the long run - as it will focus only on a small subset of manifests (since finding _any_ maintainers is already hard) and we'll not be able to keep up the repo as a whole.

**Comment by amreus** [comment]
Is there some role such as "reviewer"?  Perhaps there is a bot that could commit requests if the request is reviewed and approved by some number of "reviewers"?

**Comment by aliesbelik** [comment]
Looks like the bucket still needs more attention and/from more maintainers.

I plan to reduce my activity here (which, however, is still insufficient) so if someone has enough time and enthusiasm to fight entropy more or less regularly, now is the time.

**Comment by Fredolx** [comment]
I'd be willing to contribute my time to review PRs

**Comment by aliesbelik** [comment]
I am addressing first of all those participants who already have maintenance access in Extras bucket (since I don't have rights to add new maintainers).
New candidates should be reviewed and approved by someone with administrative privileges (@rashil2000, I suppose).

**Comment by niheaven** [comment]
Please start the review now and inform me; I will proceed with the merge soon and then appoint you as maintainers following some qualified reviews. Thank you for all your contributions!

**Comment by chaerun** [comment]
Hi Maintainers,
I'm available to help review PRs. Let me know if there are any specific guidelines or requirements I need to follow.
